Consolidate out-of-support documentation

mwoehlke-kitware opened the pull request:
Move end-of-support information to a new, consolidated document. Add a separate section to the release notes content page for "supplemental" pages (i.e. things that make sense here but aren't releases, such as the aforementioned end-of-support information). Replace old documents with a redirect and hide from the contents.

Fixes #18303. ~I'm guessing we will still need to add the EOS information for Bionic.~ Right, information for Bionic *is* added here.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/RobotLocomotion/drake/18713)
<!-- Reviewable:end -->

mwoehlke-kitware commented:
+@jwnimmer-tri for feature review, please.

The mechanism for populating the contents is somewhat crude, but is the best I managed to get working with my limited knowledge of Liquid. I'm open to better suggestions, as well as to alternate names for the tags being used.

jwnimmer-tri reviewed (commented):
+(release notes: none) for docs-only change.

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> feature.

> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/18713#-:-NN86KSA7sX_kTVV8n83:bmca8rm)* status: 2 unresolved discussions, needs at least two assigned reviewers (waiting on @mwoehlke-kitware)

___
*[`doc/_release-notes/end_of_support.md` line 12 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/18713#-NN873Zw9St-rvgSsjqK:-NN873Zw9St-rvgSsjqL:bi7ijrv) ([raw file](https://github.com/RobotLocomotion/drake/blob/0b8eea4ee0aad4c6af3f7cdea07d44b92236f28c/doc/_release-notes/end_of_support.md#L12)):*
> ```Markdown
> ## Ubuntu 18.04 (Bionic)
> 
> The last version with support for Ubuntu 16.04 is <!-- TODO vx.y.z -->:
> ```

This text isn't quite finished yet (says "16.04", and is missing a version).
___
*[`doc/_release-notes/end_of_support.md` line 20 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/18713#-NN87b_SE4YhlVxE2VaA:-NN87b_SE4YhlVxE2VaB:bhwkioz) ([raw file](https://github.com/RobotLocomotion/drake/blob/0b8eea4ee0aad4c6af3f7cdea07d44b92236f28c/doc/_release-notes/end_of_support.md#L20)):*
> ```Markdown
> The last version with support for Ubuntu 16.04 is v0.11.0:
> 
> * https://github.com/RobotLocomotion/drake/releases/tag/v0.11.0
> ```

nit For convenience, please make this a clickable hyperlink.


<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
> This text isn't quite finished yet

Right; I forgot I'd started to add the section for Binoic. 😳 Fixed.

> please make this a clickable hyperlink

Ah, I removed the gratuitously redundant `[link](link)` style links, but didn't realize Markdown wants `<>`s for automatic links. Fixed.

...and Reviewable seems to be borked right now, possibly due to a Github issue.

jwnimmer-tri reviewed (commented):
+@rpoyner-tri for platform review per schedule, please.

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/18713#-:-NN8NamT-UBuZgMWjGgx:b-z0s7k5)* status: LGTM missing from assignee rpoyner-tri(platform) (waiting on @rpoyner-tri)

___
*[`doc/_release-notes/end_of_support.md` line 12 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/18713#-NN873Zw9St-rvgSsjqK:-NN8Nd0_0a-oLDkF7G7c:b-4ku6s6) ([raw file](https://github.com/RobotLocomotion/drake/blob/0b8eea4ee0aad4c6af3f7cdea07d44b92236f28c/doc/_release-notes/end_of_support.md#L12)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

This text isn't quite finished yet (says "16.04", and is missing a version).
</blockquote></details>

Wow, we really did 1.1.0 vs 0.11.0 for our EOL releases?  That's crazy hard to read, but it's the truth!


<!-- Sent from Reviewable.io -->
